functions/New-DbaRgResourcePool.ps1

```diff
@@ -0,0 +1,197 @@
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [string[]]$ResourcePool,
+        [ValidateSet("Internal", "External")]
+        [string]$Type = "Internal",
+        [Parameter(ParameterSetName = "Internal")]
+        [ValidateRange(0, 100)]
+        [int]$MinimumCpuPercentage,
+        [Parameter(Mandatory)]
+        [ValidateRange(1, 100)]
+        [int]$MaximumCpuPercentage,
+        [Parameter(Mandatory, ParameterSetName = "Internal")]
+        [ValidateRange(1, 100)]
+        [int]$CapCpuPercentage,
+        [Parameter(ParameterSetName = "Internal")]
+        [ValidateRange(0, 100)]
+        [int]$MinimumMemoryPercentage,
+        [Parameter(Mandatory)]
+        [ValidateRange(1, 100)]
+        [int]$MaximumMemoryPercentage,
+        [Parameter(ParameterSetName = "Internal")]
+        [ValidateRange(0, 2147483647)]
+        [int]$MinimumIOPSPerVolume,
+        [Parameter(ParameterSetName = "Internal")]
+        [ValidateRange(1, 2147483647)]
+        [int]$MaximumIOPSPerVolume,
+        [Parameter(ParameterSetName = "External")]
+        [int]$MaximumProcesses,
+        [switch]$SkipReconfigure,
+        [switch]$Force,
+        [switch]$EnableException
+    )
+    process {
+        foreach ($instance in $SqlInstance) {
+            try {
+                $server = Connect-DbaInstance -SqlInstance $instance -SqlCredential $SqlCredential
+            } catch {
+                Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
+            }
+
+            foreach ($resPool in $ResourcePool) {
+                $existingResourcePool = Get-DbaRgResourcePool -SqlInstance $instance -Type $Type | Where-Object { $_.Name -eq $resPool }
+                if ($null -ne $existingResourcePool) {
+                    if ($Force) {
+                        if ($Pscmdlet.ShouldProcess($existingResourcePool, "Dropping existing resource pool '$resPool' because -Force was used")) {
+                            try {
+                                $existingResourcePool.Drop()
+                            } catch {
+                                Stop-Function -Message "Could not remove existing resource pool '$resPool' on $instance, skipping." -Target $existingResourcePool -Continue
+                            }
+                        }
+                    } else {
+                        Stop-Function -Message "Resource Pool '$resPool' already exists." -Category ResourceExists -ErrorRecord $_ -Target $existingResourcePool
+                        return
+                    }
+                }
+
+                if ($PSCmdlet.ShouldProcess($instance, "Creating resource pool '$resPool'")) {
+                    try {
+                        if ($Type -eq "External") {
+                            $splatSetDbaRgResourcePool = @{
+                                SqlInstance             = $instance
```

Comment:
You should use $server instead of $instance here.

Reply:
Will clean these up - this is leftover from local testing. In my cross-domain environment, the $server tends to return back a server name stripped of the domain so this was easier for testing.